contracts/consensus/Consensus.sol

```diff
@@ -508,15 +580,17 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
             "Block header does not match with vote message source."
         );
 
-        // Makes sure that assigned core is active.
-        address core = assignments[_metachainId];
+        bytes32 metablockHash = goToCommittedRound(_metachainId);
+
+        CoreI core = assignments[_metachainId];
         require(
```

Comment:
ok for now, but it's more of an assert than anything; the assigned core should be active

Reply:
@benjaminbollen do you mean here, that cores in `assignments` storage mapping will not keep halted/corrupted cores?